src/test/java/com/google/javascript/gents/GeneratedSyntaxTest.java

```diff
@@ -0,0 +1,70 @@
+package com.google.javascript.gents;
+
+import static com.google.javascript.clutz.DeclarationGeneratorTests.D_TS;
+import static com.google.javascript.clutz.DeclarationGeneratorTests.TS_SOURCES;
+
+import com.google.common.collect.Lists;
+import com.google.javascript.clutz.DeclarationSyntaxTest;
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.junit.Test;
+
+/**
+ * A test that uses {@code tsc} to compile the generated {@code .ts} files as a sanity check to make
+ * sure that we actually output valid TypeScript.
+ */
+public class GeneratedSyntaxTest {
+  /*
+   * Ideally we want to generate both idiomatic and valid TypeScript code. But in some cases we
+   * choose to emit idiomatic TS and sacrifice the correctness.
+   */
+  private static final String EXCLUDED_TS_ARRAY[] = {
+    "classes.ts", "module_namespace.ts", "static_methods.ts", "proto_methods.ts"
+  };
+  private static final Set<String> EXCLUDED_TS =
+      new HashSet<String>(Arrays.asList(EXCLUDED_TS_ARRAY));
+
+  private static final FilenameFilter COMPILABLE_TS_SOURCES =
+      new FilenameFilter() {
+        @Override
+        public boolean accept(File dir, String name) {
+          return TS_SOURCES.accept(dir, name) && !EXCLUDED_TS.contains(name);
+        }
+      };
+
+  @Test
+  public void testGenerated() throws Exception {
+    List<File> tsInputs =
+        TypeScriptGeneratorTests.getTestInputFiles(
+            COMPILABLE_TS_SOURCES, TypeScriptGeneratorTests.singleTestPath);
+    List<File> dtsInputs =
+        TypeScriptGeneratorTests.getTestInputFiles(D_TS, TypeScriptGeneratorTests.singleTestPath);
+    List<String> tscCommand = null;
+
+    for (File tsInput : tsInputs) {
+      tscCommand = Lists.newArrayList(DeclarationSyntaxTest.TSC.toString(), "-m", "commonjs");
+      tscCommand.addAll(DeclarationSyntaxTest.TSC_FLAGS);
+      tscCommand.add(tsInput.getPath());
+      File dtsInput = getDTS(tsInput, dtsInputs);
+      if (dtsInput != null) {
+        tscCommand.add(dtsInput.getPath());
+      }
+      DeclarationSyntaxTest.runChecked(tscCommand);
+    }
+  }
+
+  private static File getDTS(File tsInput, List<File> dtsInputs) {
```

Comment:
getDts is the preferred capitalization.

Also, mabye name getHelperDts, getAdditionalDts to make it clear these files are optional and just add some types needed for compilation (I was wonder for a bit what the purpose here).